socorro/tests/processor/rules/test_mozilla.py

```diff
@@ -1439,6 +1440,107 @@ def test_missing_crashing_thread(self, tmp_path):
         assert "topmost_filenames" not in processed_crash
 
 
+def generate_debug_id():
+    """Generates a valid debug id."""
+
+
+class TestMissingSymbolsRule:
+    def test_basic(self, tmp_path, debug_id_helper):
+        raw_crash = {}
+        dumps = {}
+        debug_id_1 = debug_id_helper.generate()
+        debug_id_2 = debug_id_helper.generate()
+        debug_id_3 = debug_id_helper.generate()
+        debug_id_4 = debug_id_helper.generate()
+        processed_crash = {
+            "json_dump": {
+                "modules": [
+                    {
+                        "filename": "libxul.dll",
+                        "version": None,
+                        "debug_id": debug_id_1,
+                        "missing_symbols": True,
+                    },
+                    {
+                        "filename": "libnss3.dll",
+                        "version": "1.0",
+                        "debug_id": debug_id_2,
+                        "missing_symbols": True,
+                    },
+                    {
+                        "filename": "mozglue.dll",
+                        "debug_id": debug_id_3,
+                        "missing_symbols": None,
+                    },
+                    {
+                        "filename": "somethingmozglue.dll",
+                        "debug_id": debug_id_4,
+                        "missing_symbols": False,
+                    },
+                ],
+            },
+        }
+        status = Status()
+
+        rule = MissingSymbolsRule()
+        rule.act(raw_crash, dumps, processed_crash, str(tmp_path), status)
+
+        assert (
+            processed_crash["missing_symbols"]
+            == f"libnss3.dll/1.0/{debug_id_2};libxul.dll/None/{debug_id_1}"
+        )
+
+    @pytest.mark.parametrize(
+        "processed_crash",
+        [
+            {},
+            {"modules": None},
+            {"modules": []},
+            {"modules": [{}]},
+            {"modules": [{"missing_symbols": True}]},
+            {"modules": [{"filename": "libxul.dll"}]},
+            {"modules": [{"filename": "libxul.dll", "version": "1.0"}]},
+        ],
+    )
+    def test_missing_things(self, tmp_path, processed_crash):
+        status = Status()
+        rule = MissingSymbolsRule()
+
+        rule.act({}, {}, processed_crash, str(tmp_path), status)
+        assert "missing_symbols" not in processed_crash
+
+    @pytest.mark.parametrize(
+        "item, expected",
+        [
+            (
+                {"filename": "libxul.so", "missing_symbols": True},
+                "libxul.so/None/" + MissingSymbolsRule.NULL_DEBUG_ID,
+            ),
+            (
+                {
+                    "filename": "libxul_2.dll",
+                    "version": "1.0",
+                    "debug_id": "51C36FAFFD214DB4A0D91D93B38336CEA",
+                    "missing_symbols": True,
+                },
+                "libxul_2.dll/1.0/51C36FAFFD214DB4A0D91D93B38336CEA",
```

Comment:
It's tricky to use the debug id helper in the parametrized options, so I hard-coded the debug id here.